src/profile-logic/profile-data.js

```diff
@@ -1140,55 +1126,34 @@ function _filterThreadByFunc(
       category: [],
       subcategory: [],
     };
+    const oldStackToNewStack = new Int32Array(stackTable.length);
 
-    const oldStackToNewStack = new Map();
-    const frameCount = frameTable.length;
-    const prefixStackAndFrameToStack = new Map(); // prefixNewStack * frameCount + frame => newStackIndex
-
-    function convertStack(stackIndex) {
-      if (stackIndex === null) {
-        return null;
-      }
-      let newStack = oldStackToNewStack.get(stackIndex);
-      if (newStack === undefined) {
-        const prefixNewStack = convertStack(stackTable.prefix[stackIndex]);
```

Comment:
with this recursivity, I feel like we were also looking up the stack info quite a lot of times (even though this was cached).
On the brighter side, we weren't looking at the stacks that are not in samples -- this could make a difference when zooming in a lot in otherwise big profiles. While in your new implementation you're looking at the stack table upfront.

I have the feeling that your new implementation is faster in most cases so I'm fine with it.

Reply:
You're right, that's a bit of a trade-off. My goal is to make all operations on the stackTable fast enough that we can eventually afford run all transforms on the full thread, so that focusing on a range and removing range filters doesn't have to re-run transforms.